beacon-chain/db/pruner/pruner.go

```diff
@@ -168,9 +184,30 @@ func (p *Service) prune(slot primitives.Slot) error {
 	// Update pruning checkpoint.
 	p.prunedUpto = pruneUpto
 
+	// Update the earliest available slot after pruning
+	log.WithField("earliestSlot", pruneUpto+1).Debug("Updating earliest available slot after pruning")
+	p.updateEarliestSlot(pruneUpto + 1)
+
 	return nil
 }
 
+// updateEarliestSlot updates the earliest available slot via the injected custody updater.
+func (p *Service) updateEarliestSlot(earliestAvailableSlot primitives.Slot) {
```

Comment:
Don't log if error. Return an wrapped error, and let the caller decide what to do.